### DISCUSSION TITLE
I'm having trouble understanding the -badname switch

### DISCUSSION THREAD

**Discussion by AltitudeApps** [rfjakob/gocryptfs#784] [General]
I did find this question: https://github.com/rfjakob/gocryptfs/discussions/657

But I still am confused as to how the `-badname` flag would even work...

The man page says:

> When gocryptfs encounters a "bad" file name (cannot be decrypted 
> or decrypts to garbage), a warning is logged and the file is hidden from the plaintext view.
> 
> With the `-badname` option, you can select "bad" file names that should still be shown in the plaintext view instead of hiding them. Bad files will get `GOCRYPTFS_BAD_NAME` appended to their name.

OK, so if the item's name decrypted to garbage, then how would that garbage name ever match any glob patterns besides `"*"` ?   Obviously, there's something I'm missing here.

The reason I'm concerned is that I'm using gocryptfs in conjunction with OneDrive.  (I'm running gocryptfs under Windows Subsystem for Linux, on a directory on my Windows NTFS drive, mounted in linux at something like `/mnt/c/Users/joeuser/OneDrive/gocryptfs/cipher`)  So I was reading the documentation looking for references to things like DropBox, SyncThing, etc., which is where I ran across the `-badname` flag.

Now I know that the `-badname` flag exists, and apparently exists because of circumstances which arise due to use of cloud sync services, I'm trying to understand why it exists.  I'm even unclear as to why anything could possibly...

Wait, I think I get it.

1. `MyFile.txt` exists in a gocryptfs plain text mount
2. file name becomes `lasdf9efokjhu` on the actual backing filesystem on the sync service
3. file gets altered on two different machines at the same time
4. the sync service now has at least two files: `lasdf9efokjhu`, and `lasdf9efokjhu.SyncConflict.Oops`
5. So the giving the argument `-badname '*.SyncConflict.Oops*'` would match against the file in the backing filesystem, and gocryptfs would present it in the FUSE mounted plaintext view as `<something-something>.GOCRYPTFS_BAD_NAME`

Did I get that right?

**Comment by rfjakob** [comment]
Yes!
